### PR TITLE
Use `https` for hyperlinked APIs by replacing the URLs

### LINKED ISSUE #573: Return secure URLs for the fields thumbnail, detail_url and related_url.
_(Framed the verbiage of the title as a feature request)_ 🙏 

## Problem

The response for search and detail requests includes insecure URLs (`http` instead of `https`) in the fields `thumbnail`, `detail_url` and `related_url`.

image

e.g.:

**Search**

https://api.openverse.engineering/v1/images/?q=flower

**Detail:**

https://api.openverse.engineering/v1/images/6c1769b0-a3e5-4dae-8a36-8531a6e1430f/

## Description

When trying to integrate Openverse with some code on the browser I ended up having to replace the scheme part of the URL for avoiding notices like ```xxxx was loaded over HTTPS, but requested an insecure resource 'http://api.openverse.engineering/v1/images/6c1769b0-a3e5-4dae-8a36-8531a6e1430f/'. This request has been blocked; the content must be served over HTTPS.`

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #573 by @oskosk

This is a proposed alternative to #574.

## Description

This PR configures Django to replace the schemes in the URLs by extending the `HyperlinkedIdentityField` with a `scheme` argument (that takes the default from the settings).

## Testing Instructions

0. Remove the `API_LINK_SCHEME` from the env files.
1. Run the services using Docker Compose: `just up`.
2. Visit the dev server at https://localhost/v1/images and observe that all URLs are https and not http.
 
## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: api/catalog/api/serializers/base.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    """
+
+    def __init__(self, scheme=settings.API_LINK_SCHEME, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.scheme = scheme
+
+    def get_url(self, *args, **kwargs):
+        url = super().get_url(*args, **kwargs)
+        return re.sub(r"\w+://", f"{self.scheme}://", url, 1)
```
Comment (AetherUnbound) [inline api/catalog/api/serializers/base.py:19]: I'd recommend anchoring this regex search, it'll be faster and will prevent anything that's later in the URL from being accidentally updated (as unlikely as that is), although I guess the `count=1` would also prevent that.
```suggestion
        return re.sub(r"^\w+://", f"{self.scheme}://", url, 1)
```

**Review by AetherUnbound — APPROVED** [review]
This looks good! It's at least an easy way to address the problem until we have a chance to restructure how we're proxying :smile:

File: api/catalog/api/serializers/base.py (hunk 1 above)
Comment (dhruvkb) [reply api/catalog/api/serializers/base.py]: Cool idea, committed in 5839c3e.

**Review by zackkrida — APPROVED** [review]
Works great! I'd propose we scrap #574 and instead follow this up with a just command, README instructions, and gitignore addition of `.pem` files while using local certificates generated with [`mkcert`.](https://github.com/FiloSottile/mkcert#installation)

I guess we'd also need some code to point django to the local certificate files, but I'll leave that to y'all 😄

**Comment by dhruvkb** [comment]
Merging this while we come up with a clean permanent solution.

**Pull request merged**